### PR TITLE
SimpleFin

### PR DESCRIPTION
**Pull request by zachwhelchel**
SimpleFin can now be configured in the UI and sync transactions, somewhat adjacent to this [issue](https://github.com/actualbudget/actual/issues/898).

There is also an app PR for these changes: https://github.com/actualbudget/actual/pull/2188

General discussion should be held on the app PR linked above.

### CODE REVIEW

File: src/app-simplefin/app-simplefin.js [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+
+const app = express();
+export { app as handlers };
+app.use(express.json());
+
+app.post('/status', async (req, res) => {
+  let configured = false;
+
+  let token = secretsService.get(SecretName.simplefin_token);
+  if (token !== null && token !== undefined && token !== 'Forbidden') {
```
Comment (joel-jeremy) [inline src/app-simplefin/app-simplefin.js:14]: These checks can be simplified to just `token != null` which check for both null and undefined.

File: src/app-simplefin/app-simplefin.js [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+        referenceDate: date,
+      },
+    ];
+    response.iban = 'thisismadeup';
+    response.institutionId = 'thisismadeup';
+    response.startingBalance = balance; // could be named differently in this use case.
+
+    let allTransactions = [];
+
+    for (let i = 0; i < account.transactions.length; i++) {
```
Comment (joel-jeremy) [inline src/app-simplefin/app-simplefin.js:100]: We can use for..in here :)

File: src/app-simplefin/app-simplefin.js [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+        balanceType: 'expected',
+        referenceDate: date,
+      },
+      {
+        balanceAmount: { amount: account.balance, currency: account.currency },
+        balanceType: 'interimAvailable',
+        referenceDate: date,
+      },
+    ];
+    response.iban = 'thisismadeup';
```
Comment (joel-jeremy) [inline src/app-simplefin/app-simplefin.js:94]: Is this intentional?

**Comment by zachwhelchel** [comment]
@joel-jeremy @youngcw @MatissJanis this PR is ready as well now.

**Pull request merged**